infrastructure/templates/public-api/components/postgresqlDatabase.bicep

```diff
@@ -0,0 +1,153 @@
+@description('Specifies the Subscription to be used.')
+param subscription string
+
+@description('Specifies the location for all resources.')
+param location string
+
+//Specific parameters for the resources
+@description('Server Name for Azure Database for PostgreSQL')
+param serverName string = 'metadata'
+
+@description('Database administrator login name')
+@minLength(1)
+param administratorLoginName string
+
+@description('Database administrator password')
+@minLength(8)
+@secure()
+param administratorLoginPassword string
+
+@description('Azure Database for PostgreSQL sku name, typically, tier + family + cores, e.g. Standard_D4s_v3 ')
+param skuName string
+
+@description('Azure Database for PostgreSQL Storage Size ')
+param storageSizeGB int
+
+@description('Azure Database for PostgreSQL Autogrow setting')
+param autoGrowStatus string
+
+@description('Azure Database for PostgreSQL pricing tier')
+@allowed([
+  'Burstable'
+  'GeneralPurpose'
+  'MemoryOptimized'
+])
+param skuTier string = 'Burstable'
+
+@description('PostgreSQL version')
+@allowed([
+  '11'
+  '12'
+  '13'
+  '14'
+  '15'
+])
+param postgresqlVersion string = '14'
+
+@description('PostgreSQL Server backup retention days')
+param backupRetentionDays int = 7
+
+@description('Geo-Redundant Backup setting')
+param geoRedundantBackup string = 'Disabled'
+
+@description('The name of the Key Vault to store the connection strings')
+param KeyVaultName string
+
+//Passed in Tags
+param departmentName string = 'Public API'
+param environmentName string = 'Development'
+param solutionName string = 'API'
+param subscriptionName string = 'Unknown'
+param costCentre string = 'Unknown'
+param serviceOwnerName string = 'Unknown'
+param dateProvisioned string = utcNow('u')
+param createdBy string = 'Unknown'
+param deploymentRepo string = 'N/A'
+param deploymentScript string = 'N/A'
+
+// Variables and created data
+var metadataDatabaseName = '${subscription}-postgre-${serverName}'
+var pythonDbConnectionString = 'database=<database>, user=${administratorLoginName}@${metadataDatabase.name}, host=${metadataDatabase.name}${environment().suffixes.sqlServerHostname}, password=${administratorLoginPassword}, port=5432'
+var adoNetDbConnectionString = 'Server=${metadataDatabase.name}${environment().suffixes.sqlServerHostname};Database=<database>;Port=5432;User Id=${administratorLoginName}@${metadataDatabase.name};Password=${administratorLoginPassword};'
```

Comment:
In Bicep, strings are marked with singled quotes, and must be declared on a single line.
Using concat throws an error with the linter.

Use string interpolation instead of the concat function.
The following example fails this test because the concat function is used.

Reply:
You could disable the linter in this situation: https://learn.microsoft.com/en-us/azure/azure-resource-manager/bicep/linter#silencing-false-positives